### PR TITLE
refactor: mock files and codes

### PR DESCRIPTION
**Pull request by mabdh**
These 3 points are being done here:
- [x]  Refactor global variables so every initialization done in the server creation
- [x]  Decouple highly coupled package & services
- [x]  Refactor files in domain package to its respective domain and organize the mock files better

One additional issue is solved
- [x] Integrate coverall to the repo

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1,24 +1,59 @@
...
-			return &cfg, nil
+			return cfg, nil
 		}
-		return nil, err
+		return cfg, err
 	}
-	return &cfg, nil
+	cfg.Cortex.PrometheusAlertManagerConfigYaml = promAMConfigYamlString
+	cfg.Cortex.PrometheusAlertManagerHelperTemplate = promAMHelperTemplateString
+	cfg.SirenService.Port = cfg.Port
```
[the remark this answers is not in the document]
Comment (mabdh) [reply config/config.go:36]: Our current [config](https://github.com/odpf/siren/blob/main/config/config.yaml#L12) structure has `port` on the top level of config. On the other hand it also has `siren_service` config and inside it has only `host`. 

If I can guess we use `siren_service` config only for our siren cli and port only for server. I think we can have better structure for this. We should only have 1 config with host and port.
```
siren_service:
    host: localhost
    port: 8080
```
I am doing this for backward compatibility purpose, we can refactor it to the better solution describe above later in another changes. Actually we can do that in this PR but I am afraid it is too much changes and it is harder to be reviewed.

Sorry I forgot to add comment in the code explaining this.

File: cli/rule.go [hunk 2]
```diff
@@ -24,20 +24,20 @@ type variables struct {
 	Name  string `yaml:"name"`
 	Value string `yaml:"value"`
 }
-type rule struct {
+type ruleStruct struct {
```
[the remark this answers is not in the document]
Comment (mabdh) [reply cli/rule.go:27]: by `renaming the github.com/odpf/siren/core/rule`, do you mean aliasing it in this file only? or renaming the whole package ?

[the remark this answers is not in the document]
Comment (mabdh) [reply cli/rule.go:27]: anyway, we can rename it into anything, any suggestion for the naming @rahmatrhd ?

[the remark this answers is not in the document]
Comment (rahmatrhd) [reply cli/rule.go:27]: yes I meant aliasing. let's keep the package name as `rule` only

File: config/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rahmatrhd) [reply config/config.go:37]: we can also add deprecated flag into the field itself https://go.dev/blog/godoc#:~:text=Sometimes%20a%20struct,about%20the%20deprecation.

File: cli/rule.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mabdh) [reply cli/rule.go]: @rahmatrhd sorry after checking the code, aliasing imported rule package would require us to alias all imported rule package in other files inside `cli` package. I was thinking to alias it to `ruledomain` but that makes it inconsistent with other imported domains.

What about we just remove the `ruleStruct`?

```
type ruleYaml struct {
	ApiVersion        string `yaml:"apiVersion"`
	Entity            string `yaml:"entity"`
	Type              string `yaml:"type"`
	Namespace         string `yaml:"namespace"`
	Provider          string `yaml:"provider"`
	ProviderNamespace string `yaml:"providerNamespace"`
	Rules             map[string]struct {
		Template  string      `yaml:"template"`
		Enabled   bool        `yaml:"enabled"`
		Variables []variables `yaml:"variables"`
	} `yaml:"rules"`
}
```

File: config/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mabdh) [reply config/config.go:37]: Added deprecated comment.

File: cli/rule.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rahmatrhd) [reply cli/rule.go]: ah i see, yes I think `ruleYaml` inside `cli` package makes sense 👍

**Review by rahmatrhd — APPROVED** [review]

File: cli/rule.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mabdh) [reply cli/rule.go]: Thanks @rahmatrhd !

**Pull request merged**